### PR TITLE
🎨 Making service start progress UI user readable

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- Renamed progress element details to make it more user friendly for non developer users.
- Items are now displayed in the order they are done in the backend

Screenshot 2024-02-05 at 10 41 36

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by pcrespov — COMMENTED** [review]
thx

**Review by elisabettai — COMMENTED** [review]
Following the suggestion from @mguidon [here](https://z43.fogbugz.com/f/cases/190406/New-feedback-Rename-Reorder-Progress-Bars), I'd just have these steps:
1. Installing platform services
2. Increasing system capacity
3. Installing software
4. Downloading files

For our (dev) understanding
1. corresponds to pulling sidecar (which is normally very fast)
2. corresponds to autoscaling 
3. corresponds to pulling images
4. corresponds to download outputs+state+inputs (users don't care about what exactly is being downloaded, but it would be nice if they understand that the more/bigger files they have, the slower this step will be.

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Comment by GitHK** [comment]
> Following the suggestion from @mguidon [here](https://z43.fogbugz.com/f/cases/190406/New-feedback-Rename-Reorder-Progress-Bars), I'd just show these steps:
> 
> 1. Installing platform services...
> 2. Increasing system capacity...
> 3. Installing software...
> 4. Downloading files...
> 
> For our (dev) understanding
> 
> 1. corresponds to pulling sidecar (which is normally very fast)
> 2. corresponds to autoscaling
> 3. corresponds to pulling images
> 4. corresponds to download outputs+state+inputs (users don't care about what exactly is being downloaded, but it would be nice if they understand that the more/bigger files they have, the slower this step will be.

@elisabettai we decided to go in favour of these new ones after a chat with MaG. It is not clear to me if you want changes to the names.

**Comment by GitHK** [comment]
> Following the suggestion from @mguidon [here](https://z43.fogbugz.com/f/cases/190406/New-feedback-Rename-Reorder-Progress-Bars), I'd just show these steps:
> 
> 1. Installing platform services...
> 2. Increasing system capacity...
> 3. Installing software...
> 4. Downloading files...
> 
> For our (dev) understanding
> 
> 1. corresponds to pulling sidecar (which is normally very fast)
> 2. corresponds to autoscaling
> 3. corresponds to pulling images
> 4. corresponds to download outputs+state+inputs (users don't care about what exactly is being downloaded, but it would be nice if they understand that the more/bigger files they have, the slower this step will be.

@elisabettai I'm not sure how to deal with the removal of the sections without crashing the FE. I would leave that part to someone with more Javascript experience. If it is really required to hide those parts.

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js [hunk 1]
```diff
@@ -206,29 +200,35 @@ qx.Class.define("osparc.data.model.NodeProgressSequence", {
       const clusterUpScalingPBar = this.__clusterUpScalingPBar = this.self().createProgressBar();
       this.__sequenceLoadingPage.add(clusterUpScalingPBar);
 
-      const pullingInputsTitle = this.__pullingInputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Pulling inputs..."));
-      this.__sequenceLoadingPage.add(pullingInputsTitle);
+      const pullingSidecarTitle = this.__pullingSidecarTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Setting up key components..."));
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:203]: I'd rephrase to: "Installing platform services..."

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js [hunk 2]
```diff
@@ -206,29 +200,35 @@ qx.Class.define("osparc.data.model.NodeProgressSequence", {
...
 
-      const pullingOutputsTitle = this.__pullingOutputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Pulling outputs..."));
+      const pullingOutputsTitle = this.__pullingOutputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Downloading outputs..."));
       this.__sequenceLoadingPage.add(pullingOutputsTitle);
 
       const pullingOutputsPBar = this.__pullingOutputsPBar = this.self().createProgressBar();
       this.__sequenceLoadingPage.add(pullingOutputsPBar);
 
-      const pullingStateTitle = this.__pullingStateTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Pulling state..."));
+      const pullingStateTitle = this.__pullingStateTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Retrieving your work..."));
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:215]: Downloading your files...

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js [hunk 3]
```diff
@@ -206,29 +200,35 @@ qx.Class.define("osparc.data.model.NodeProgressSequence", {
...
       this.__sequenceLoadingPage.add(pullingStatePBar);
 
-      const pullingImagesTitle = this.__pullingImagesTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Pulling images..."));
+      const pullingImagesTitle = this.__pullingImagesTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Installing software..."));
       this.__sequenceLoadingPage.add(pullingImagesTitle);
 
       const pullingImagesPBar = this.__pullingImagesPBar = this.self().createProgressBar();
       this.__sequenceLoadingPage.add(pullingImagesPBar);
+
+      const pullingInputsTitle = this.__pullingInputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Downloading inputs..."));
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:227]: Downloading your inputs...

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js [hunk 4]
```diff
@@ -206,29 +200,35 @@ qx.Class.define("osparc.data.model.NodeProgressSequence", {
...
+      const pullingSidecarTitle = this.__pullingSidecarTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Setting up key components..."));
+      this.__sequenceLoadingPage.add(pullingSidecarTitle);
 
-      const pullingInputsPBar = this.__pullingInputsPBar = this.self().createProgressBar();
-      this.__sequenceLoadingPage.add(pullingInputsPBar);
+      const pullingSidecarPBar = this.__pullingSidecarPBar = this.self().createProgressBar();
+      this.__sequenceLoadingPage.add(pullingSidecarPBar);
 
-      const pullingOutputsTitle = this.__pullingOutputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Pulling outputs..."));
+      const pullingOutputsTitle = this.__pullingOutputsTitle = this.self().createTitleAtom(qx.locale.Manager.tr("Downloading outputs..."));
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:209]: Downloading your outputs...

**Comment by elisabettai** [comment]
> > Following the suggestion from @mguidon [here](https://z43.fogbugz.com/f/cases/190406/New-feedback-Rename-Reorder-Progress-Bars), I'd just show these steps:
> > 
> > 1. Installing platform services...
> > 2. Increasing system capacity...
> > 3. Installing software...
> > 4. Downloading files...
> > 
> > For our (dev) understanding
> > 
> > 1. corresponds to pulling sidecar (which is normally very fast)
> > 2. corresponds to autoscaling
> > 3. corresponds to pulling images
> > 4. corresponds to download outputs+state+inputs (users don't care about what exactly is being downloaded, but it would be nice if they understand that the more/bigger files they have, the slower this step will be.
> 
> @elisabettai I'm not sure how to deal with the removal of the sections without crashing the FE. I would leave that part to someone with more Javascript experience. If it is really required to hide those parts.

I see @GitHK , I thought that the file you edited was the only place where those steps were defined and that was relatively easy to test if you were to break the FE. 

I've also added some proposal for rephrasing some of the messages.

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 2 above)
Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:215]: Q1: why make it more technical?
Q2: inputs and outputs also contain files. I think it becomes a bit inconsistent.

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 3 above)
Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:227]: 👍

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 4 above)
Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:209]: 👍

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 1 above)
Comment (elisabettai) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:203]: I thought "Setting up key components...." is very vague. The user might wonder: are they setting up my s4l model, simulation, etc...?

While "Installing platform services...." suggests something they should not care about. 

General comment: in my opinion, users only care about their data, models, simulations, code, etc... and I think it's important that they understand that if they have big files for their models, simulations, etc, they need to be patient while we download them.

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 2 above)
Comment (elisabettai) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:215]: Mmh, true, but also outputs are part of "your work".

What about "Downloading your workspace"? Once they have used it one/two types, they should know what the "workspace" folder is.

If you don't like it, "Retrieving your work" also works.

Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:215]: While outputs are part of your work. The term work is a more generic concept, unlike workspace which is a bit more technical. Not all services will call the state folder as worksapce

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 1 above)
Comment (elisabettai) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:203]: My point was exactly that a too vague definition might induce users to think something else is happening (and something else is normally something they know). 😉

**Review by elisabettai — APPROVED** [review]

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 1 above)
Comment (mrnicegyu11) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:203]: Both are fine, with a gun to my head I prefer Elis suggestion

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 2 above)
Comment (jsaq007) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:215]: Personally I wouldn't use downloading as it sounds like you are downloading them to your computer not the workspace. Perhaps something like:

Retrieving assets... (Then include inputs and outputs in this why have extra details)

Then something like 

Generating workspace...

**Review by jsaq007 — APPROVED** [review]
